shapeshift/web
# feat: add Li.Fi 25% gasLimit buffer in getApprovalTxData

## Description

Does what it says on the box - makes it so the gasLimit we use has the same buffer as in https://github.com/lifinance/sdk/pull/148
If we were using LiFi's approval utils, we would automagically get the same values, however:
1. we're using the LiFi SDK "raw" values and doing the signing/broadcasting ourselves given we don't have a signer. That's as opposed to using the utils which would get the same values as Jumper.
2. We're not even using Li.Fi values at all for approvals - approvals are different than other "steps", we're using our own values here - both for display purposes and to actually build the Tx

Hence, we conform to LiFi SDK's buffer. Not sure why there is such a buffer in the first place (ethers estimation being off? An oracle making the gas limit consistently underestimated for certain tokens?) but point approvals works on Li.Fi for BSC tokens using the default BSC JSON-RPC MM endpoint, but doesn't in our app.


## Pull Request Type


## Issue (if applicable)

N/A

## Risk



Worst case scenario we're overestimating gas limit now.

The real risk in testing this is MM being dumb and using cached previously manually input values despite it clearly stating that it's using "app-suggested values". This will require tests from multiple peep and/or testing from fresh MM instances (i.e new Chrome profile with freshly installed MM)

## Testing



- Make sure you remove any custom BSC network you have in MM (i.e not MM default's one with Binance's RPC endpoint). Alternatively, just modify the endpoint to be `https://bsc-dataseed.binance.org/`
- Ensure Li.Fi approvals for BSC tokens are going through

### Engineering



- ^
- With the multi-hop flag on, the actual Tx gasLimit should be 25% higher than currently for Li.Fi approval Txs only - other Txs shouldn't be affected

### Operations



- ^

## Screenshots (if applicable)

<img width="629" alt="image" src="https://github.com/shapeshift/web/assets/17035424/d2d2d5be-f43b-4d56-bac5-05b5e630f408">
<img width="608" alt="image" src="https://github.com/shapeshift/web/assets/17035424/153b2eb0-1b8e-4e05-981b-586753b08571">
